### ISSUE TITLE
Error while training: validation batch NaN

### ISSUE THREAD

**Issue by pocaha** [zylo117/Yet-Another-EfficientDet-Pytorch#357]
Hey zylo117, 
when training from scratch, I get always this error for validation batches
Val. Epoch: 0/500. Classification loss: nan. Regression loss: nan. Total loss: nan

**Comment by zylo117** [comment]
it's usally overfitting and exploding gradients.

**Comment by pocaha** [comment]
Thank you very much for this information. Though this simple I was having a hard time to understnd wht is going on. After changing my optimizer from SGD to Adamw, i don't have this Problem anymore. However, after 7 epochs (I had for epoch 7 a total loss of 1.6 and for the validation total loss 1.37), all of a sudden my total loss has risen to 94578458.0. It went 4 orders down than two orders up and now its stays in that high range (5 orders of magnitude). Can you kindly explain me why this is happening?

**Comment by zylo117** [comment]
mmm...
I think it's in machine learning 101.
You should probably google 'exploding loss'
